File: src/main/java/me/itzisonn_/meazy/runtime/environment/ModifieredEnvironment.java
package me.itzisonn_.meazy.runtime.environment;

import me.itzisonn_.meazy.parser.modifier.Modifier;

import java.util.Set;

public interface ModifieredEnvironment extends Environment {
    /**
     * @return This class environment's modifiers
     */
    Set<Modifier> getModifiers();

    default boolean hasModifier(Modifier modifier) {
        return getModifiers().contains(modifier);
    }
}

File: src/main/kotlin/me/itzisonn_/meazy/runtime/environment/ModifieredEnvironment.kt
package me.itzisonn_.meazy.runtime.environment

import me.itzisonn_.meazy.runtime.data.modifier.Modifier

interface ModifieredEnvironment : Environment {
    /**
     * @return This class environment's modifiers
     */
    val modifiers: Set<Modifier>

    fun hasModifier(modifier: Modifier) = modifier in modifiers
}
